Rubberduck.Core/Navigation/CodeExplorer/CodeExplorerReferenceViewModel.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System;
+using System.Windows.Media.Imaging;
+using Rubberduck.AddRemoveReferences;
+using Rubberduck.Resources.CodeExplorer;
+using Rubberduck.VBEditor;
+
+namespace Rubberduck.Navigation.CodeExplorer
+{
+    public class CodeExplorerReferenceViewModel : CodeExplorerItemViewModel
+    {
+        private readonly ReferenceModel _reference;
+
+        public CodeExplorerReferenceViewModel(CodeExplorerReferenceFolderViewModel parent, ReferenceModel reference)
+        {
+            Parent = parent;
+            _reference = reference;
+        }
+
+        public override string NameWithSignature => $"{_reference.Name} ({_reference.Version})";
+        public override string Name => _reference.Description + Environment.NewLine + _reference.FullPath;
+        public override CodeExplorerItemViewModel Parent { get; }
+        public override QualifiedSelection? QualifiedSelection => null;
+
+        public override BitmapImage CollapsedIcon => GetIcon();
+        public override BitmapImage ExpandedIcon => GetIcon();
+
+        public int? Priority => _reference.Priority;
+        public bool Locked => _reference.IsBuiltIn;
+
+        private BitmapImage GetIcon()
+        {
+            if (_reference.Status.HasFlag(ReferenceStatus.Broken))
+            {
+                GetImageSource(CodeExplorerUI.BrokenReference);
+            }
+
+            return _reference.IsBuiltIn ? GetImageSource(CodeExplorerUI.LockedReference) : GetImageSource(CodeExplorerUI.Reference);
+        }
```

Comment:
That would probably be better, but it more or less matches how it's handled in the other `CodeExplorerItemViewModel` implementations. Ideally this would be in xaml, but I think that would require changing all of the other existing icon customization code in the CE. TBH, the entire tree-view probably needs an overhaul - there's enough shared code with the metrics window that it may warrant its own control. That a bit outside the scope of the PR, so I'm inclined to punt on this until we do that.

Reply:
Ok, I only saw the diff. If this is identical for the other properties, it is definetly out of scope for this PR.